app/frontend/packs/components/checkbox_search_filter.js

```diff
@@ -0,0 +1,203 @@
+    })
+  }
+
+  getAllCheckboxes() {
+    return this.checkboxesContainer.querySelectorAll('.govuk-checkboxes__item')
+  }
+
+  getAllVisibleCheckboxes() {
+    return [].filter.call(this.getAllCheckboxes(), function(el) {
+      return el.style.display == 'block'
+    })
+  }
+
+  getAllVisibleCheckedCheckboxes() {
+    return [].filter.call(this.getAllVisibleCheckboxes(), function(el) {
+      return el.querySelector('.govuk-checkboxes__input').checked
+    })
+  }
+
+  getTextBoxHtml() {
+    var container_id = this.container.id
+    var label = document.createElement('label')
+    label.setAttribute('for', container_id + '-checkbox-filter__filter-input')
+    label.className = 'govuk-label govuk-visually-hidden'
+    label.innerHTML = this.params.textBox.label
+
+    var input = document.createElement('input')
+    input.className = 'app-checkbox-filter__filter-input govuk-input'
+    input.setAttribute('id', container_id + '-checkbox-filter__filter-input')
+    input.setAttribute('type','text')
+    input.setAttribute('aria-describedby', container_id + '-checkboxes-status')
+    input.setAttribute('aria-controls', container_id + '-checkboxes')
+    input.setAttribute('autocomplete', 'off')
+    input.setAttribute('spellcheck', 'false')
+
+    return [label, input]
+  }
+
+  getVisibleCheckboxes() {
+    var visibleCheckboxes = [].filter.call(this.checkboxes, (el) => this.isCheckboxInView(el))
+
+    // add an extra checkbox, if the label of the first is too long it collapses onto itself
+    visibleCheckboxes.push(this.checkboxes[visibleCheckboxes.length-1])
+    return visibleCheckboxes
+  }
+
+  isCheckboxInView(element) {
+    var checkbox = element
+    var initialOptionContainerHeight = this.checkboxesContainer.style.height
+    var optionListOffsetTop = this.checkboxesInnerContainer.offsetTop
+    var distanceFromTopOfContainer = checkbox.offsetTop - optionListOffsetTop
+    return distanceFromTopOfContainer < initialOptionContainerHeight
+  }
+
+  onTextBoxKeyUp(e) {
+    var ENTER_KEY = 13
+    if (e.keyCode === ENTER_KEY) {
+      e.preventDefault()
+    } else {
+      this.filterCheckboxes()
+    }
+  }
+
+  sanitizeText(text) {
+    text = text.replace(/&/g, 'and')
+    text = text.replace(/[’',:–-]/g, '') // remove punctuation characters
+    text = text.replace(/[.*+?^${}()|[\]\\]/g, '\\$&') // escape special characters
+    return text.trim().replace(/\s\s+/g, ' ').toLowerCase() // replace multiple spaces with one
+  }
+
+  setContainerHeight(height) {
+    this.checkboxesContainer.style.height = height
+  }
+
+  setup() {
+    this.setupStatusBox()
+    this.setupHeading()
+    if (this.checkboxes.length >= 15) {
+      this.setupTextBox()
+    }
+    this.setupHeight()
+  }
+
+  setupHeading() {
+    var element = document.createElement('p')
+    element.className = 'app-checkbox-filter__title'
+    element.setAttribute('aria-hidden', 'true')
+
+    this.heading = element
+    this.container.prepend(this.heading)
+  }
```

Comment:
This is required in order for the input field to be rendered (look at line 176), but it doesnt actually render any text, as the component's label already exists and is rendered from the  just before the subject container - I believe that is handled by the existing filters JS.